src/SwedbankPay/Api/Client/ClientVersion.php

```diff
@@ -172,27 +174,25 @@ private function tryGetVersionNumberFromComposerJson(&$version) : bool
       */
     private function tryGetVersionNumberFromComposerLock(&$version) : bool
     {
-        $composerLock = null;
+        $paths = [];
 
-        if (!$this->tryReadComposerLock($composerLock)) {
-            return false;
-        }
+        // Standard path of composer.lock
+        $path = $this->getComposerPath() . DIRECTORY_SEPARATOR . 'composer.lock';
+        $paths[] = $path;
 
-        if (isset($composerLock['packages'])) {
-            $packages = $composerLock['packages'];
-            foreach ($packages as $package) {
-                if (!isset($package['name']) ||
-                    $package['name'] !== 'swedbank-pay/swedbank-pay-sdk-php') {
-                    continue;
-                }
-
-                if (isset($package['version'])) {
-                    $version = $package['version'];
-
-                    if ($version !== null && !empty($version)) {
-                        return true;
-                    }
-                }
+        // Alternate path of composer.lock
+        $pathDirs = explode(DIRECTORY_SEPARATOR, $path);
+        array_splice($pathDirs, -2);
+        $pathDirs[] = 'composer.lock';
+        $paths[] = implode(DIRECTORY_SEPARATOR, $pathDirs);
```

Comment:
This code will only append the nearest parent of the standard path rather than traverse recursively through the file directory path. As part of tryReadComposerLock() it was able to call itself with the parent path as argument, making it recursive. Why was this code moved?